src/SmartComponents/Recs/Details.js

```diff
@@ -53,7 +53,12 @@ const OverviewDetails = ({ isImmutableTabOpen }) => {
     data: recAck = {},
     isFetching: recAckIsFetching,
     refetch: recAckRefetch,
-  } = useGetRecAcksQuery({ ruleId });
+  } = useGetRecAcksQuery(
+    { ruleId },
+    {
+      skip: !!ruleId,
```

Comment:
According to my investigation. it rather happens always when the component mounts initially. So, it does not matter if a user navigates to an invalid URL or not. This can be verified by running the master branch locally and opening any recommendation.

Reply:
I am not sure why `id` is undefined on the initial mount. Will dig more and better fix that if possible